projects/dsp-ui/src/lib/core/session.service.ts

```diff
@@ -0,0 +1,215 @@
+                        sysAdmin: sysAdmin,
+                        projectAdmin: projectAdmin
+                    }
+                };
+
+                // update localStorage
+                localStorage.setItem('session', JSON.stringify(session));
+            },
+            (error: ApiResponseError) => {
+                localStorage.removeItem('session');
+                console.error(error);
+            }
+        );
+    }
+
+    /**
+     * Validate intern session and check knora api credentials if necessary.
+     * If a json web token exists, it doesn't mean that the knora api credentials are still valid.
+     *
+     * @returns boolean
+     */
+    isSessionValid(): boolean {
+        // mix of checks with session.validation and this.authenticate
+        const session = JSON.parse(localStorage.getItem('session'));
+
+        const tsNow: number = this.setTimestamp();
+
+        if (session) {
+
+            this.updateDspApiConnection(session.user.jwt);
+
+            // check if the session is still valid:
+            if (session.id + this.MAX_SESSION_TIME <= tsNow) {
+                // the internal (dsp-ui) session has expired
+                // check if the api credentials are still valid
+
+                this.dspApiConnection.v2.auth.checkCredentials().subscribe(
+                    (response: ApiResponseData<CredentialsResponse>) => {
+                        // the knora api credentials are still valid
+
+                        // update the session.id
+                        session.id = tsNow;
+
+                        localStorage.setItem('session', JSON.stringify(session));
+
+                        return true;
+                    },
+                    (error: ApiResponseError) => {
+                        // a user is not authenticated anymore!
+                        this.destroySession();
+                        return false;
+                    }
+                );
+
+            } else {
+                // the internal (dsp-ui) session is still valid
+                return true;
+            }
+        } else {
+            // no session found; update knora api connection with empty jwt
+            this.updateDspApiConnection();
+            return false;
+        }
+    }
+
+    /**
+     * update the session storage
+     * @param jwt
+     * @param username
+     *
+     * @returns boolean
+     */
+    updateSession(jwt: string, username: string) {
+        if (jwt && username) {
+            this.setSession(jwt, username, 'username');
+        }
+    }
+
+    /**
+     * Destroy session by removing the session from local storage
+     *
+     */
+    destroySession() {
+        localStorage.removeItem('session');
+    }
+
+    /**
+     * Update the dsp-api-config and dsp-api-connection of @dasch-swiss/dsp-js
+     *
+     * @param  {string} jwt?
+     */
+    private updateDspApiConnection(jwt?: string) {
+        this.dspApiConfig.jsonWebToken = (jwt ? jwt : '');
```

Comment:
Isn't it enough to set the jwt on the `DspApiConfigToken`?

Reply:
private methods should start with an underscore

Reply:
> Isn't it enough to set the jwt on the DspApiConfigToken?

There is this property `jsonWebToken` in the dspApiConfig provided by dsp-js-lib. I thought it's a good idea to assign. When it's not needed and we do not have to set it, we can delete this line. But why do we have this `jsonWebToken` property in dspApiConfig?

Reply:
the jwt is set on the `KnoraApiConfig`, see https://github.com/dasch-swiss/knora-api-js-lib/blob/bc169c0403b19d81096be81d871ef3c12ff29549/src/api/endpoint.ts#L35-L47

Reply:
Yes, dspApiConfig is of type KnoraApiConfig. So, we don't have to set it?

Reply:
dspApiConfig is initiated in AppInitService and can be used in the ui-lib. Described in the ui-lib README: 
https://www.npmjs.com/package/@dasch-swiss/dsp-ui#setup

Reply:
For some reason there are problems with the config token, see  https://github.com/dasch-swiss/knora-ui-ng-lib/pull/118. Whenever I try to use it ina component or a service, I get an error (dependency injection).

Reply:
I saw that there's a missing part in CoreModule. Please have a look at the old knora-ui core module:
https://github.com/dasch-swiss/knora-ui/blob/master/projects/knora/core/src/lib/core.module.ts

```typescript
    static forRoot(kuiConfig: KuiConfig): ModuleWithProviders {
        // get the app environment configuration here
        // console.log('KuiCoreModule - forRoot - config: ', config);
        return {
            ngModule: KuiCoreModule,
            providers: [
                { provide: KuiConfigToken, useValue: kuiConfig }
            ]
        };
    }
```

Reply:
I will fix it...

Reply:
could you try that on https://github.com/dasch-swiss/knora-ui-ng-lib/pull/118?

Reply:
You should mention the type, see https://angular.io/guide/migration-module-with-providers:

```
@NgModule({...})
export class MyModule {
  static forRoot(config: SomeConfig): ModuleWithProviders<SomeModule> {
    return {
      ngModule: SomeModule,
      providers: [
        {provide: SomeConfig, useValue: config }
      ]
    };
  }
}
```

Reply:
> could you try that on #118?

Yes I will

Reply:
> private methods should start with an underscore

Resolved in [6c9bc68](https://github.com/dasch-swiss/knora-ui-ng-lib/pull/84/commits/6c9bc68bd1e0eb715973d7ee48df8675202883c3)

Reply:
> Isn't it enough to set the jwt on the `DspApiConfigToken`?

I agree now. Resolved in 75c1377

Reply:
> I saw that there's a missing part in CoreModule. Please have a look at the old knora-ui core module:
> https://github.com/dasch-swiss/knora-ui/blob/master/projects/knora/core/src/lib/core.module.ts
 
I'm sorry, but my answer was nonsense. This is a leftover from the old knora-ui setup and doesn't make sense anymore. I have tested it in #118 but have already undone it.

Reply:
Part of this conversation is resolved in PR #118